pkg/cloud/services/eks/roles.go

```diff
@@ -17,59 +17,43 @@ limitations under the License.
 package eks
 
 import (
-	"encoding/json"
 	"fmt"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/iam"
 	"github.com/pkg/errors"
 
-	infrav1 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3"
 	ekscontrolplanev1 "sigs.k8s.io/cluster-api-provider-aws/controlplane/eks/api/v1alpha3"
+	infrav1exp "sigs.k8s.io/cluster-api-provider-aws/exp/api/v1alpha3"
+	eksiam "sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/eks/iam"
 	"sigs.k8s.io/cluster-api-provider-aws/pkg/record"
 )
 
-// TrustRelationshipPolicyDocument represesnts an IAM policy docyment
-type TrustRelationshipPolicyDocument struct {
-	Version   string
-	Statement []StatementEntry
-}
-
-// ToJSONString converts the document to a JSON string
-func (d *TrustRelationshipPolicyDocument) ToJSONString() (string, error) {
-	b, err := json.Marshal(d)
-	if err != nil {
-		return "", err
+// NodegroupRolePolicies gives the policies required for a nodegroup role
+func NodegroupRolePolicies() []string {
+	return []string{
+		"arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+		"arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy", // TODO get rid of this with SA?
```

Comment:
Amazon recommends [using IAM roles for service accounts](https://github.com/aws/containers-roadmap/issues/766) and then removing this Policy from worker nodes because it's quite permissive. But doing this requires setting up an OIDC provider which I suppose is out of scope for this PR.

Reply:
Cool. Could we update the text to something like
`//TODO: Can remove when CAPA supports provisioning of OIDC web identity federation with service account volume token projection`

We should chat about this, because I would like to do this eventually, fits in with the multi-tenancy work as well.